File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let numbersWanted = {}

    for(let i =0; i<nums.length; i++) {
        const res = target - nums[i]
        if(Object.keys(numbersWanted).includes(String(nums[i])))
            return [i, numbersWanted[nums[i]]]
        else
            numbersWanted[res] = i
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> diff = new HashMap<>();
        
        for(int i=0; i<nums.length; i++) {
            int d = target - nums[i];
            
            if(diff.containsKey(d)) {
                return new int[] {i, diff.get(d)};
            } else {
                diff.put(nums[i], i);
            }
        }
        
        return new int[] {-1, -1};
    }
}